cmd/internal/persist_output.go

```diff
@@ -22,10 +27,47 @@ into a versionsed secret or kube native secret using flags specified to this com
 			return errors.Errorf("persist-output command failed. cf-operator-namespace flag is empty.")
 		}
 
-		return extendedjob.PersistOutput(namespace)
+		// hostname of the container is the pod name in kubernetes
+		podName, err := os.Hostname()
+		if err != nil {
+			return errors.Wrapf(err, "failed to fetch pod name.")
+		}
+		if podName == "" {
+			return errors.Wrapf(err, "pod name is empty.")
+		}
+
+		// Authenticate with the cluster
+		clientSet, versionedClientSet, err := authenticateInCluster()
+		if err != nil {
+			return err
+		}
+
+		po := extendedjob.NewPersistOutputInterface(namespace, podName, clientSet, versionedClientSet, "/mnt/quarks")
+
+		return po.PersistOutput()
 	},
 }
 
 func init() {
 	utilCmd.AddCommand(persistOutputCmd)
 }
+
+// authenticateInCluster authenticates with the in cluster and returns the client
+func authenticateInCluster() (*kubernetes.Clientset, *versioned.Clientset, error) {
```

Comment:
Could we reuse the package `"code.cloudfoundry.org/cf-operator/pkg/kube/config" like:

https://github.com/cloudfoundry-incubator/cf-operator/blob/08944f9767e2196e101f74139dd9291bb2cae690/cmd/internal/root.go#L52-L58

Reply:
Done.